hashicorp/waypoint
# ui: build and release tabs tables view

## TODOs
- [x] Update UIs
  - [x] Add table view
  - [x] Update `AppItem` row components
- [x] Update where the link is located on the table row

## Remaining TODOs for the future
- [ ] [Add popover to `Deployment` column](https://github.com/hashicorp/waypoint/issues/2740)
- [ ] [Add `Deploy now` button back w/ functionality](https://github.com/hashicorp/waypoint/issues/2445)
- [ ] (Maybe) update the way the links function in the rows
- [ ] Add filtering/sorting functionality

## Notes
- Flight icons currently does not have animated icons, so the loading icon for when a build/release is in progress is currently the `delay` icon and will be updated when the correct icon is available

## How to test
1. Pull down branch `git checkout ui/build-release-tables`
2. `yarn start` to test with Mirage data, or start up Waypoint server and `ember serve local`
3. Check the builds and releases table views.

## Screenshots

### Empty State

<img width="1437" alt="Screen Shot 2021-12-23 at 2 23 53 PM" src="https://user-images.githubusercontent.com/60155296/147283954-d4cb706b-165d-4e8f-ae5b-b38faaeb5312.png">

<img width="1437" alt="Screen Shot 2021-12-23 at 2 23 59 PM" src="https://user-images.githubusercontent.com/60155296/147283956-1ba7821b-65fe-44a9-8bb2-be3e76998842.png">

### Build Table
<img width="1439" alt="Screen Shot 2021-12-23 at 2 27 32 PM" src="https://user-images.githubusercontent.com/60155296/147284223-b5eaf6a4-b72e-4930-a978-3fc8d3f2669d.png">

### Release Table
<img width="1436" alt="Screen Shot 2021-12-23 at 2 27 24 PM" src="https://user-images.githubusercontent.com/60155296/147284234-8abed56a-89da-460f-b65f-c5ecf681834e.png">
